### PR TITLE
feat(docs): pull in content from craft-parts

### PR DESCRIPTION
**Pull request by tigarmo**
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

File: requirements-doc.txt [hunk 1]
```diff
@@ -8,7 +8,7 @@ colorama==0.4.6
 craft-application==1.2.1
 craft-archives==1.1.3
 craft-cli==2.5.1
-craft-parts==1.26.1
+craft-parts @ git+https://github.com/canonical/craft-parts@plugin-doc-details
```
Comment (tigarmo) [inline requirements-doc.txt:11]: reminder to fix this

**Review by syu-w — APPROVED** [review]

**Review by lengau — CHANGES_REQUESTED** [review]
Great job overall!

Could you add an example for the python plugin? Before doing hacky work with classic snaps & python I wouldn't have understood what's needed for getting the dependencies working.

File: docs/.wordlist.txt
[the remark this answers is not in the document]
Comment (tigarmo) [reply docs/.wordlist.txt:1]: This is something else that I want to bundle in the craft-parts docs package too

File: docs/reference/plugins/python_plugin.rst [hunk 2]
```diff
@@ -1,2 +1,20 @@
...
+Since none of the bases that are available for rocks contain a default Python
+installation, including a Python interpreter in Rockcraft projects is mandatory.
+The plugin also requires the ``venv`` module to create the virtual environment
+where Python packages are installed at build time.
+
+The easiest way to do this is to include the ``python3-venv`` package in the
+``stage-packages`` of the part that uses the Python plugin. This will pull in
+the default Python interpreter for the ``build-base``, like Python 3.10 for
+Ubuntu 22.04. However, other versions can be used by explicitly declaring them,
+such as ``python3.12-venv`` for projects that use the Deadsnakes ppa.
```
[the remark this answers is not in the document]
Comment (tigarmo) [reply docs/reference/plugins/python_plugin.rst]: that's a good idea - added a yaml snippet that includes python 3.12 from deadsnakes

**Review by lengau — APPROVED** [review]
Looks great, thanks!

File: docs/reference/plugins/python_plugin.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (lengau) [reply docs/reference/plugins/python_plugin.rst]: Thanks!

**Pull request merged**